### PR TITLE
docs(contributing, conventions): proofread and add additional info

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #3818

## Summary
- Added additional information to the contributing and documentation convention.
- Fixed outdated info

### CODE REVIEW

File: CONTRIBUTING.md
Comment (geospatialem) [inline CONTRIBUTING.md:14]: Since there is an action requested with the statement, could we add a sub-bullet for the "Before working on an issue..." sentence?

Also, added a suggestion for the action's clarity below.

Perhaps:
- Working on the issues marked as help wanted. There is also a good first issue label if you are just getting started.
    - Before working ~on an issue, make sure to~ comment on [the issue] ~~it~~ and ask for the action items. Sometimes there is additional context needed that is not specified in the issue.

File: CONTRIBUTING.md
Comment (geospatialem) [inline CONTRIBUTING.md:78]: A suggestion for the second sentence to be less on the reader:

You [can familiarize yourself with]  ~~should read through~~ the label descriptions ~~to familiarize yourself with what they~~ [to understand what they] signify.

Comment (geospatialem) [reply CONTRIBUTING.md:78]: Should we mention these sentences earlier on in the doc since labels are mentioned prior to this section?

File: CONTRIBUTING.md
Comment (geospatialem) [inline CONTRIBUTING.md:97]: Should these two extensions be mentioned if we're referencing no specific editor above?

File: conventions/Documentation.md
Comment (geospatialem) [inline conventions/Documentation.md:16]: Additional comma?:

- Use backticks (`` ` ``) for the names of slots, events, properties[,] CSS variables, and component names (e.g. `calcite-button` instead of calcite-button).

File: conventions/Documentation.md
Comment (geospatialem) [inline conventions/Documentation.md:22]: Some ending sentences?:

- Use "Accessible" instead of "Aria" or "a11y" language[.]
- Verify slots and properties/attributes don't use the text "optional" in their descriptions[.]

File: conventions/Documentation.md
Comment (geospatialem) [inline conventions/Documentation.md:63]: ...during every `next` version[,] where[as] the SDK site is only...

**Review by geospatialem — COMMENTED** [review]
Nice updates - great inclusion of the style guide notes!  💯 A few small considerations.

File: CONTRIBUTING.md
Comment (benelan) [reply CONTRIBUTING.md:97]: Thanks for all the feedback! I'll make the changes. Yeah I was on the fence about this too but opted for keeping them since they were already there. I'll remove them.

File: conventions/Documentation.md
Comment (jcfranco) [inline conventions/Documentation.md:45]: The `storiesOf` API [is considered legacy](https://github.com/storybookjs/storybook/blob/master/lib/core/docs/storiesOf.md#storiesof-legacy-api). Stories should follow [CSF](https://storybook.js.org/docs/html/api/csf).

File: CONTRIBUTING.md
Comment (jcfranco) [inline CONTRIBUTING.md:16]: I think this should point to the intro page: https://stenciljs.com/docs/introduction

File: CONTRIBUTING.md
Comment (jcfranco) [inline CONTRIBUTING.md:92]: Nitpick: `StyleLint` ➡️ `Stylelint`.

File: CONTRIBUTING.md
Comment (jcfranco) [inline CONTRIBUTING.md:179]: Just an idea, but it'd be great to mention somewhere that developers can modify the conventional commit right before merging if additional edits were needed.

**Review by jcfranco — APPROVED** [review]
📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖
📖🤩📖📖📖📖📖🤩📖🤩🤩🤩🤩📖🤩📖📖📖📖📖🤩📖🤩🤩🤩🤩📖🤩📖📖📖📖📖🤩📖🤩🤩🤩🤩📖🤩🤩🤩🤩📖🤩📖📖📖📖📖🤩📖🤩🤩🤩🤩📖
📖🤩📖📖📖📖📖🤩📖🤩📖📖🤩📖🤩📖📖📖📖📖🤩📖🤩📖📖🤩📖🤩📖📖📖📖📖🤩📖🤩📖📖📖📖🤩📖📖📖📖🤩📖📖📖📖📖🤩📖🤩📖📖🤩📖
📖🤩📖📖🤩📖📖🤩📖🤩🤩🤩🤩📖🤩📖📖🤩📖📖🤩📖🤩🤩🤩🤩📖🤩📖📖🤩📖📖🤩📖🤩🤩🤩📖📖🤩🤩🤩📖📖🤩📖📖🤩📖📖🤩📖🤩🤩🤩🤩📖
📖🤩📖🤩📖🤩📖🤩📖🤩📖📖🤩📖🤩📖🤩📖🤩📖🤩📖🤩📖📖🤩📖🤩📖🤩📖🤩📖🤩📖🤩📖📖📖📖🤩📖📖📖📖🤩📖🤩📖🤩📖🤩📖🤩📖📖🤩📖
📖📖🤩📖📖📖🤩📖📖🤩📖📖🤩📖📖🤩📖📖📖🤩📖📖🤩📖📖🤩📖📖🤩📖📖📖🤩📖📖🤩🤩🤩🤩📖🤩🤩🤩🤩📖📖🤩📖📖📖🤩📖📖🤩📖📖🤩📖
📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖📖

It'd be nice to update the writing stories section, but we can tackle this separately.

File: CONTRIBUTING.md
Comment (benelan) [reply CONTRIBUTING.md:179]: Good idea, I can just copy and paste what the script tells them on the CLI https://github.com/Esri/calcite-components/blob/5cd7119f6eb575756d5232fd621ae0441b2c0b8f/support/checkSquashMergeableBranch.ts#L27

Comment (benelan) [reply CONTRIBUTING.md:179]: Oh nevermind I misread, yes thats also I good idea lol

**Pull request merged**
